app/jobs/order_cycle_webhook_job.rb

```diff
@@ -4,6 +4,11 @@
 class OrderCycleWebhookJob < ApplicationJob
   def perform(order_cycle_id, event)
     order_cycle = OrderCycle.find(order_cycle_id)
+    return if already_notified?(order_cycle, event)
```

Comment:
This is prone to a race condition. Maybe we do it in one query like `OrderCycle.lock.opened_unnotified.find`?

Reading this I'm getting a bit confused about the different jobs. Maybe they need an explanation in a comment at the top. Who's triggering them when and what do they trigger.